testsuite/tests/compaction/test_freelist_free.ml

```diff
@@ -0,0 +1,19 @@
+(* TEST
+*)
+
+(* tests that we correctly empty the shared pool's freelist. This requires a
+    bunch of garbage to be generated, a major cycle and two compactions to
+    test. *)
```

Comment:
This test doesn't make much sense on its own because it doesn't check that the memory is actually released.
```suggestion
(* tests that we correctly empty the shared pool's freelist. This requires a
    bunch of garbage to be generated, a major cycle and two compactions to
    test. If we can do that without segfault, we're good. *)
```